src/mca/plm/base/plm_base_jobid.c

```diff
@@ -96,13 +97,18 @@ int prrte_plm_base_create_jobid(prrte_job_t *jdata)
         found = false;
         for (i=1; i < INT16_MAX; i++) {
             ptr = NULL;
-            pjid = PRRTE_CONSTRUCT_LOCAL_JOBID(PRRTE_PROC_MY_NAME->jobid, i);
+            pjid = PRRTE_CONSTRUCT_LOCAL_JOBID(PRRTE_PROC_MY_NAME->jobid, prrte_plm_globals.next_jobid);
```

Comment:
Urr..that isn't right, is it? You are looking at the same location every time thru the loop, so you will never find an open slot.

Reply:
There is an increment on like 122 that advances it for the function. Then an increment at the bottom of the for loop to cycle it around.